### DISCUSSION TITLE
memray without python?

### DISCUSSION THREAD

**Discussion by lrbison** [bloomberg/memray#554] [General]
Memray has a lot of great interfaces and features!  It got me thinking, would it be possible to use the native mode to track allocations in a c application that isn't python at all?  I tried something similar using follow-fork and subprocess.run, but I found the binary I executed had no information tracked in memray.  It seems like you have all the tools, perhaps all that is missing is an LD_PRELOAD with some stub malloc/free to stand in for PyMalloc.  Is this something you've tried or configured?

**Comment by pablogsal** [comment]
HI @lrbison and thanks for your message!

Certainly `memray` can work with all the Python pieces stripped, but that's quite a lot of work to do. Understand that `memray` it's a very specialised profiler that is quite coupled to ensure it provides the best experience for Python applications in a way that doesn't require to use LD_PRELOAD (so you can activate and deactivate, attach...etc). This means that while it's certainly possible (all the tools are there) it's quite a lot of work to strip that out into it's own thing, so it's unlikely that we will do it.

**Comment by lrbison** [comment]
I see.  thank you!  I guess I'll stick to python only code for now.
